### ISSUE TITLE
get_columns_in_relation does not include columns from external tables in Redshift

### ISSUE THREAD

**Issue by aiguofer** [fishtown-analytics/dbt#2753]
### Describe the bug
get_columns_in_relation does not include columns from external tables in Redshift

### Steps To Reproduce
- Set up a Spectrum tables as a source in your project
- Use `adaptor.get_columns_in_relation(source(....))` in a macro

### Expected behavior
All the column names should be returned. 

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [x] redshift
- [ ] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
❯ dbt --version
installed version: 0.17.0
   latest version: 0.18.0

Your version of dbt is out of date! You can find instructions for upgrading here:
https://docs.getdbt.com/docs/installation

Plugins:
  - bigquery: 0.17.0
  - snowflake: 0.17.0
  - redshift: 0.17.0
  - postgres: 0.17.0

```

**The operating system you're using:**
MacOS Catalina

**The output of `python --version`:**
Python 3.7.7

**Comment by aiguofer** [comment]
Slightly related to #2289 but solution is different.

**Comment by jtcohen6** [comment]
I like it!

**Comment by jtcohen6** [comment]
Closed by #2754
